### PR TITLE
MOR 1947: morgen tasks obsidian plugin hide ids in reading mode

### PR DESCRIPTION
**Pull request by lukebarnard1**
I will not squash these commits as the first is clearly unrelated and is purely related to formatting the codebase.

The second one is the one I would like a review of.

This fixes the problem where IDs were not hidden/replaced in:
 - reading mode
 - queries
 - Kanban boards (inline/card footer mode)

Screenshots previewing "replace with emoji" mode:

This first one is the outlier because it doesn't make much sense to replace the Kanban plugin ID card footer chip with an emoji, so it's just hidden in this mode.
CleanShot 2024-09-23 at 15 09 09

CleanShot 2024-09-23 at 15 09 13
CleanShot 2024-09-23 at 15 09 20
CleanShot 2024-09-23 at 15 09 43

### CODE REVIEW

File: main.ts [hunk 1]
```diff
@@ -2,21 +2,50 @@ import { App, Plugin, PluginSettingTab, Setting } from 'obsidian';
...
+		 * plugins. There are some cases where task IDs do not have an obvious
+		 * containing span element that can be easily replaced/hidden. These
+		 * are handled by the MorgenTasksPostProcessor:
+		 *   - Tasks plugin has a bug that causes IDs that have text after them
+		 *     in the line to NOT get a container span, which is normally handled
+		 *     by the style.css rules
+		 *   - Kanban plugin by default shows tasks in "inline" mode, which is
+		 *     an alternative to "Move task data to card footer". It does not
+		 *     wrap the ID in a containing span.
+		 */
```
Comment (marcoancona) [inline main.ts:45]: Thanks, great context!

**Review by marcoancona — APPROVED** [review]
LGTM. I wonder if it would be a valid addition to wrap the entire postProcess method in a try/catch just in case there is a hidden issue with this logic. I don't know what happens if a postprocessor throws in Obsidian, but we don't want to cause any issue.

**Comment by lukebarnard1** [comment]
>LGTM. I wonder if it would be a valid addition to wrap the entire postProcess method in a try/catch just in case there is a hidden issue with this logic. I don't know what happens if a postprocessor throws in Obsidian, but we don't want to cause any issue.

That's a good point indeed, I will test what happens.

**Comment by lukebarnard1** [comment]
Yeah if an error happens here, it will cause the document to not render any content in the reading view that had the error.

I will wrap it so that it gives up after an error.

**Pull request merged**